tests/unit/CertificateGenerationService.unitTest.ts

```diff
@@ -960,28 +989,293 @@ describe("Certificate Generation Service", () => {
-            .catch(() => {
-              expect(logSpy).toHaveBeenCalledWith(
-                  "Welsh certificate generation deactivated via environment variable set to TRUE"
-              );
-              logSpy.mockClear();
-            });
+          const logSpy = jest.spyOn(console, "warn");
+
+          const isWelsh = certGenSvc.isGlobalWelshFlagEnabled(featureFlags);
+          expect(isWelsh).toBeFalsy();
+          expect(logSpy).toHaveBeenCalledWith("Unable to translate any test results: global Welsh flag disabled.");
+          logSpy.mockClear();
+          jest.resetAllMocks();
+        });
       });
-    });
 
+      context("test isTestResultFlagEnabled method", () => {
+        context("PASS Flag", () => {
+          const event = cloneDeep(queueEventPass);
+          const testResult: any = JSON.parse(event.Records[0].body);
+
+          it("should return true when flag is enabled", async () => {
+            const featureFlags = {
+              welshTranslation: {
+                enabled: true,
+                translatePassTestResult: true,
+                translatePrsTestResult: false,
+                translateFailTestResult: false,
+              },
+            };
+            mockGetProfile.mockReturnValueOnce(Promise.resolve(featureFlags));
+
+            const certGenSvc = new CertificateGenerationService(
+                null as any,
+                new LambdaService(new Lambda())
+            );
+
+            const isWelsh = certGenSvc.isTestResultFlagEnabled(featureFlags, testResult.testTypes.testResult);
+            expect(isWelsh).toBeTruthy();
+            jest.resetAllMocks();
+          });
+
+          it("should return false when flag is disabled and log relevant warning", async () => {
+            const featureFlags = {
+              welshTranslation: {
+                enabled: true,
+                translatePassTestResult: false,
+                translatePrsTestResult: false,
+                translateFailTestResult: false,
+              },
+            };
+            mockGetProfile.mockReturnValueOnce(Promise.resolve(featureFlags));
+
+            const certGenSvc = new CertificateGenerationService(
+                null as any,
+                new LambdaService(new Lambda())
+            );
+
+            const logSpy = jest.spyOn(console, "warn");
+
+            const isWelsh = certGenSvc.isTestResultFlagEnabled(featureFlags, testResult.testTypes.testResult);
+            expect(isWelsh).toBeFalsy();
+            expect(logSpy).toHaveBeenCalledWith(`Unable to translate for test result: pass flag disabled`);
+            logSpy.mockClear();
+            jest.resetAllMocks();
+          });
+        });
+
+        context("PRS Flag", () => {
+          const event = cloneDeep(queueEventPRS);
+          const testResult: any = JSON.parse(event.Records[0].body);
+
+          it("should return true when flag is enabled", async () => {
```

Comment:
should not allow Welsh translation for a test result, when the prs flag is disabled and log a warning